controller/api/util/api_utils.go

```diff
@@ -5,13 +5,12 @@ import (
 	"errors"
 	"fmt"
 	"strings"
-	"time"
 
+	netPb "github.com/linkerd/linkerd2/controller/gen/common/net"
 	"github.com/linkerd/linkerd2/pkg/k8s"
 	pb "github.com/linkerd/linkerd2/viz/metrics-api/gen/viz"
```

Comment:
Agreed about Tap. OTOH, that `Resource` type is pretty generic, so we can also think about moving it into `proto/common`.

Reply:
The end goal should definitely be to get rid of all references to viz in the main control plane.  Probably we need a new internal `Resource` struct (or perhaps there's something in API-machinery we can use) and these `BuildResource` functions should return that instead of a protobuf struct.  Then the Tap CLI will need to be able to convert from that resource type into a protobuf resource struct.

Reply:
@alpeb mind opening up a tracking issue for this if one doesn't already exist?

Reply:
I've created #5589 :+1: